clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_5_0.xsd

```diff
@@ -0,0 +1,1380 @@
+        <xs:attribute name="server-socket-bindings" type="tns:list">
+            <xs:annotation>
+                <xs:documentation>
+                    List of outbound-socket-bindings of Hot Rod servers to connect to.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="socket-timeout" type="xs:long" default="60000">
+            <xs:annotation>
+                <xs:documentation>
+                    Enable or disable SO_TIMEOUT on socket connections to remote Hot Rod servers with the specified timeout, in milliseconds.
+                    A timeout of 0 is interpreted as an infinite timeout.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="tcp-no-delay" type="xs:boolean" default="true">
+            <xs:annotation>
+                <xs:documentation>
+                    Enable or disable TCP_NODELAY on socket connections to remote Hot Rod servers.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="tcp-keep-alive" type="xs:boolean" default="false">
+            <xs:annotation>
+                <xs:documentation>
+                    Configures TCP keepalive on the TCP stack.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="value-size-estimate" type="xs:int" default="512">
+            <xs:annotation>
+                <xs:documentation>
+                    This hint allows sizing of byte buffers when serializing and deserializing values, to minimize array resizing.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="remote-cache">
+        <xs:attribute name="name" type="xs:string">
+            <xs:annotation>
+                <xs:documentation>
+                    Defines a remote cache name.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="invalidation-near-cache">
+        <xs:attribute name="max-entries" type="xs:int" default="-1">
+            <xs:annotation>
+                <xs:documentation>
+                    Defines the maximum number of elements to keep in the near cache.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="connection-pool">
+        <xs:attribute name="exhausted-action" type="tns:exhausted-action" default="WAIT">
+            <xs:annotation>
+                <xs:documentation>
+                    Specifies what happens when asking for a connection from a server's pool, and that pool is exhausted.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="lifo" type="xs:boolean" default="true">
```

Comment:
You know my general philosophy on boolean attributes...
What do you think of replacing this with strategy="FIFO|LIFO" or pool="QUEUE|STACK"?